app/controllers/groups_controller.rb

```diff
@@ -71,7 +71,12 @@ def delete_user
   def add_user
     if current_user.admin? || @group.admin?(current_user)
       user = User.find(params[:user_id])
-      @group.add_user(user.id) if user.present?
+      begin
+        expiration_date = expiration_date_from_param
+      rescue ArgumentError
```

Comment:
it's likely this PR will be refactored, but please avoid using exception as control flow whenever possible.